[Dev] Workflow for Augmenting GenVal with Additional/Selectable Algorithm Support

KennyBDykes opened the pull request:
This PR introduces Redis-backed queue handling to the GenVal application, enabling asynchronous job processing between services as part of our initial workflow integration.
@DewSecGitHub 

Key Changes

Added GenValWorker background service to process Redis jobs
Defined GenValRequestJob and GenValResponseJob for structured message handling
Supports get-supported-algorithms operation with scalable pattern for future job types
Added additional methods to get supported algorthims so that they are more readable 
Responses are pushed back to a separate queue (veep:responses) for consumption by Veep application


Redis is being used as an initial mechanism to implement our queuing workflow because:
It’s simple to integrate and quick to test locally
It offers sufficient performance for our current throughput
The pattern is scalable and can later support distributed deployments
Easy to change if we want to use another mechanism like RabbitMq 

Future Considerations
Remote access: We’ll need to expose Redis to external applications securely (e.g., via remote host or tunneling)
Containerization: We plan to containerize the GenVal service and Redis setup using Docker for consistent development and deployment environments

🧪 How to Test

Start Redis locally (brew services start redis)
Run the GenVal app
Push a request to genval:requests and observe response on veep:responses

<img width="1620" alt="Screenshot 2025-05-30 at 1 56 36 PM" src="https://github.com/user-attachments/assets/796f6e8a-6300-418b-88e1-caa6f454d4cb" />

DewSecGitHub commented on gen-val/samples/GenValAppRunner/src/Program.cs:
```diff
@@ -36,49 +44,73 @@ static Program()
+                  .UseConsoleLifetime()
+                  .Build();
+                await host.RunAsync();
+                return 0;
```
Is this intended as a status / error return value ?  Can we use constants anytime we are setting static values?

KennyBDykes replied:
Yes, returning 0 typically signals successful execution (exit code 0), and non-zero values indicate errors. I can use named constants or enums for exit codes it can improve readability and maintainability.

KennyBDykes commented on _config/Directory.Packages.props:
```diff
@@ -52,6 +52,7 @@
     <PackageVersion Include="Serilog.Settings.Configuration" Version="8.0.2" />
     <PackageVersion Include="Serilog.Sinks.File" Version="6.0.0" />
     <PackageVersion Include="Serilog.Sinks.Console" Version="4.1.1" />
+    <PackageVersion Include="StackExchange.Redis" Version="2.8.37" />
```
ok

DewSecGitHub reviewed (changes requested):
@KennyBDykes it seems we are adding Job Queue support to the GenVal-side and just simply using Redis rather than Orleans as NIST does.  Original design was to have the Queues on the Swift-side and then feed the Swift-Queues<->GenVal (1:1) process accordingly.  Is there an issue or limitation with GenVal in taking that approach ?  One of the reasons I would like to keep the queues on the Swift-side is that the code will need to be maintainable for the life of the product.  We are trying hard to limit the use of C#/.Net in this platform or to push PR request back up to NIST where it makes sense for their implementation.

I have fully reviewed this PR and made a couple other comments, but was hoping to resolve my question above before merging this into `main`.  Thanks!

DewSecGitHub commented:
@KennyBDykes it also looks like this branch is out-of-date / out-of-sync with main and I need for you to resolve that to ensure I don't misapply logic to the conflict resolution.

KennyBDykes commented:
@DewSecGitHub 
Overview of Changes
Removed Redis queue handling and background polling worker (GenValWorker).
Deleted GenValRequestJob and GenValResponseJob classes since job queuing is no longer needed.
Added GenValApiController to expose supported GenVal operations via REST endpoints.
Refactored Program.cs to support ASP.NET Core Web API with proper routing and Swagger support.

Notes for Review
Swagger UI is available at: http://localhost:5000/swagger for local testing.

<img width="1616" alt="Screenshot 2025-05-30 at 6 13 28 PM" src="https://github.com/user-attachments/assets/bd71f26f-b167-4d1a-8002-118fa7359390" />

DewSecGitHub reviewed (approved):
Reviewed.  Approved.  Merging.
